src/architecture.c

```diff
@@ -322,37 +352,65 @@ static void arithm(arch_core* core)
 
 static void mov_imm(arch_core* core)
 {
-
+	arch_uint eff_data = ((cond_imm_data)core->regs.ir.data).addr;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg; // Or use dreg??
+  	*help_get_reg(core, eff_reg) = eff_data;
 }
 
 static void add_imm(arch_core* core)
 {
-	
+	arch_uint eff_data = ((arith_data)core->regs.ir.data).src1 + ((arith_data)core->regs.ir.data).src2;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
+  	*help_get_reg(core, eff_reg) = eff_data;
 }
 
 static void mul_imm(arch_core* core)
 {
-
+	arch_uint eff_data = ((arith_data)core->regs.ir.data).src1 * ((arith_data)core->regs.ir.data).src2;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
+  	*help_get_reg(core, eff_reg) = eff_data;
 }
 
 static void div_imm(arch_core* core)
 {
-
+	arch_uint eff_data = ((arith_data)core->regs.ir.data).src1 / ((arith_data)core->regs.ir.data).src2;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
+  	*help_get_reg(core, eff_reg) = eff_data;
 }
 
 static void load_imm(arch_core* core)
 {
-
+	arch_uint eff_data
+	if (((cond_imm_data)core->regs.ir.data).dreg == 0)
+	{
+		eff_data = ((cond_imm_data)core->regs.ir.data).addr;
+	}
+	else
+	{
+		eff_data = ((cond_imm_data)core->regs.ir.data).dreg;
+	}
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg; // Or use dreg??
```

Comment:
You would use `breg`. `dreg` is supposed to be 0 whenever `addr` represents an immediate value.